### ISSUE TITLE
Clean up after TF 2 and new NLU architecture

### ISSUE THREAD

**Issue by tabergma** [RasaHQ/rasa#5351]
Clean up/Tackle remaining open tasks from https://github.com/RasaHQ/rasa/pull/5266

- [ ] Add docstrings and types to all methods, classes (especially in policies and NLU components)
- [ ] Split up long methods in `DIETClassifier (e.g. `_create_model_data`, `_predict_label`) and in `hf_transformers.py` (e.g. `_load_model`)
- [ ] In `DIETClassifier`: Throw warning or error on not enough labels? (should be consistent with other classifiers, e.g. `SklearnIntentClassifier`)
- [ ] Check if the spacy version check in `SpacyTokenizer` is still needed
- [ ] Consider moving the `tensorflow` package out of the `utils` package

**Comment by m-vdb** [comment]
@ricwo is there anything to discuss here or should we move it to backlog?

**Comment by Ghostvv** [comment]
@samsucik I feel like the models part is addressed by your PR: https://github.com/RasaHQ/rasa/pull/7589/files/0aaf2646bed3b515c50ebb82e716e6c1450c3f3c..6003daee3941229ec56ecb545d7ce9a7a789532e

@dakshvar22 is anything additional needs to be done for hf transformers?

things like checking spacy version and moving packages around are engineering tasks

**Comment by samsucik** [comment]
@Ghostvv just by reading the issue, I think my PR addresses the models part only partly, not fully...
